### PR TITLE
Updated to use squeue, poll multiple jobs

### PR DESCRIPTION
**Pull request by charles-cowart**

### CODE REVIEW

File: sequence_processing_pipeline/Job.py [hunk 1]
```diff
@@ -191,20 +212,81 @@ def _system_call(self, cmd, allow_return_codes=[], callback=None):
...
+        # job-id.
+
+        def query_slurm(job_ids):
+            # internal function query_slurm encapsulates the handling of
+            # squeue.
+            result = self._system_call(f"squeue -t all -j {','.join(job_ids)}"
+                                       " -o '%F,%A,%T'")
+
+            if result['return_code'] != 0:
+                raise ExecFailedError(result['stderr'])
```
Comment (antgonza) [inline sequence_processing_pipeline/Job.py:236]: What about having a wait/try for 3 times before raising an error; things can fail for many reasons.

**Review by antgonza — COMMENTED** [review]
Just one comment/question.

File: sequence_processing_pipeline/Job.py (hunk 1 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/Job.py:236]: okay. I'll try adding a sleep and repeat to code.  testing now on qiita-rc.

**Comment by charles-cowart** [comment]
This job does not contain new tests nor modifies any existing tests since its behavior depends interaction with a Slurm job scheduler. This code was developed and tested on qiita-rc however and found to work correctly, however.

**Pull request merged**